### PR TITLE
[FIX] project: make tests demo data independent

### PR DESCRIPTION
**Pull request by MissingNoShiny**
The test `test_search_project_root_id` fails when demo data are not installed. This PR makes it demo data independent.

Task-3410352

### CODE REVIEW

File: addons/project/tests/test_project_base.py [hunk 1]
```diff
@@ -199,6 +199,7 @@ def test_change_project_or_partner_company(self):
             self.project_pigs.company_id = company_1
 
     def test_search_project_root_id(self):
+        self.env['project.project'].search([]).allow_milestones = True
```
Comment (xavierbol) [inline addons/project/tests/test_project_base.py:202]: I am not sure it is really useful, the test seems to use the project created and maybe the one of the parent task and that's it, no? 🧐

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left one comment and after that it will look good to me. 🙂

File: addons/project/tests/test_project_base.py (hunk 1 above)
Comment (MissingNoShiny) [reply addons/project/tests/test_project_base.py:202]: The problem is that by default, without demo data, all other projects will have `allow_milestones` to False, so the assert at line 225 will fail

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+

**Pull request closed without merging**